server/mqtt.go

```diff
@@ -3484,7 +3496,7 @@ func (sess *mqttSession) untrackPubRel(pi uint16) (jsAckSubject string) {
 func (sess *mqttSession) deleteConsumer(cc *ConsumerConfig) {
 	sess.mu.Lock()
 	sess.tmaxack -= cc.MaxAckPending
-	sess.jsa.sendq.push(&mqttJSPubMsg{subj: sess.jsa.prefixDomain(fmt.Sprintf(JSApiConsumerDeleteT, mqttStreamName, cc.Durable))})
+	sess.jsa.deleteConsumer(mqttStreamName, cc.Durable, true)
```

Comment:
Added for code re-use. Also, when I first implemented the noWait-deleteConsumer similarly to this, it started failing `TestMQTTSessionNotDeletedOnDeleteConsumerError` because the expected error (can't delete the consumer) was missing in the server log. Using the idiomatic `NewRequest` restored the error, so I figured it's a better way of sending the request.